sources/api/ecs-settings-applier/src/main.rs

```diff
@@ -0,0 +1,158 @@
+#[macro_use]
+extern crate log;
+
+use std::{env, process};
+use std::fs;
+use std::path::{Path, PathBuf};
+use serde::{Serialize};
+use snafu::{ResultExt};
+use error::SettingsApplierError;
+
+const DEFAULT_API_SOCKET: &str = "/run/api.sock";
+const DEFAULT_ECS_CONFIG_PATH: &str = "/etc/ecs/ecs.config.json";
+const VARIANT_ATTRIBUTE_NAME: &str = "bottlerocket.variant";
+const VERSION_ATTRIBUTE_NAME: &str = "bottlerocket.version";
+
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all="PascalCase")]
+struct ECSConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cluster: Option<String>,
+
+    #[serde(skip_serializing_if = "std::collections::HashMap::is_empty")]
+    instance_attributes: std::collections::HashMap<String, String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    privileged_disabled: Option<bool>,
+}
+
+// Returning a Result from main makes it print a Debug representation of the error, but with Snafu
+// we have nice Display representations of the error, so we wrap "main" (run) and print any error.
+// https://github.com/shepmaster/snafu/issues/110
+fn main() {
+    if let Err(e) = run() {
+        eprintln!("{}", e);
+        process::exit(1);
+    }
+}
+
+fn run() -> Result<()> {
+    let args = parse_args(env::args());
+
+    // Get all settings values for config file templates
+    debug!("Requesting settings values");
+    let settings = schnauzer::get_settings(&args.socket_path).context(error::SettingsError)?;
+
+    debug!("settings = {:#?}", settings.settings);
+    let ecs = settings.settings.and_then(|s| s.ecs);
+    let ecs = if let Some(x) = ecs {
+        x
+    } else {
+        return Err(error::SettingsApplierError::ModelError);
+    };
+    let cluster = ecs.cluster.as_ref();
+    let privileged_disabled = ecs.allow_privileged_containers.map(|s| !s);
+    let mut config = ECSConfig{
+        cluster: cluster.map(|s| s.clone()),
+        instance_attributes: std::collections::HashMap::new(),
+        privileged_disabled,
+    };
+    match settings.os {
+        None => {}
+        Some(os) => {
+            config.instance_attributes.insert(VARIANT_ATTRIBUTE_NAME.to_string(), os.variant_id);
+            config.instance_attributes.insert(VERSION_ATTRIBUTE_NAME.to_string(), os.version_id.to_string());
+        }
+    }
+    match ecs.instance_attributes.as_ref() {
+        None => {}
+        Some(attributes) => {
+            for (key, value) in attributes {
+                config.instance_attributes.insert(key.to_string(), value.to_string());
+            }
+        }
+    }
```

Comment:
I don't think `as_ref` is needed here:
```suggestion
    match ecs.instance_attributes {
        None => {}
        Some(attributes) => {
            for (key, value) in attributes {
                config.instance_attributes.insert(key.to_string(), value.to_string());
            }
        }
    }
```

Actually I think this and the match above (for `settings.os`) can be simplified to something like the following:
```suggestion
    if let Some(attributes) = ecs.instance_attributes {
        for (key, value) in attributes {
                config.instance_attributes.insert(key.to_string(), value.to_string());
        }
    }
```

Reply:
Done (by Jake's commit)